### PR TITLE
Add ML component v.1.0.0-rc.0

### PR DESCRIPTION
**Pull request by mstopa**

### CODE REVIEW

File: ml/src/app.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    return [(label, str(round(score, 4))) for (label, score) in zip(
+        sort_by_ranking(labels), 
+        sort_by_ranking(scores)
+    )]
+
+app = Flask(__name__)
+
+health = HealthCheck()
+
+def classify_works():
```
Comment (cansubmarinesswim) [inline ml/src/app.py:43]: If classification does not make sense, does that mean that the `service is broken`? The service might be fine, the model could just be wrong. If it returns `False` the service still works (kinda). Maybe separate endpoints for those two cases?

File: ml/demo_client.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        )
+    parser.add_argument(
+        '-p', 
+        '--port', 
+        dest='port', 
+        type=str,
+        default="8080"
+        )
+    parser.add_argument(
+        '--health-check',
```
Comment (cansubmarinesswim) [inline ml/demo_client.py]: Maybe just `--health-check`?

File: ml/.dockerignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+*
+!./version.py
+!./setup.py
+!./README.md
+!./requirements.txt
+!src/app.py
```
Comment (cansubmarinesswim) [inline ml/.dockerignore]: `!src` should be enough.

File: ml/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
+FROM python:3.8-slim
+
+WORKDIR /opt/posts_sentiment_analyser
+
+COPY . .
```
Comment (cansubmarinesswim) [inline ml/Dockerfile]: If the `requirements.txt` file only contains pypi packages (like in this case) a good practice is to:
1. First, copy only `requirements.txt`
2. Install packages from `requirements.txt`
3. Copy all the rest.

It allows for better caching - in your case, if you modify any file before another image build, the cache miss will occur, and all the packages will be downloaded again and installed again. If you copy requirements.txt first, and install all the packages right after - the cache miss will occur only when `requirements.txt` is changed.

Its mentioned in [best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy)

File: ml/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+RUN python -m pip install --upgrade pip setuptools wheel ;
+RUN pip install torch --extra-index-url https://download.pytorch.org/whl/cpu
+RUN pip install -r requirements.txt
+
+RUN pip install .
```
Comment (cansubmarinesswim) [inline ml/Dockerfile:13]: Try to squeeze most of it into one layer (but have in mind the stuff from `requirements.txt` comment)

File: ml/demo_client.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        '--host', 
+        dest='host', 
+        type=str,
+        default="127.0.0.1"
+        )
+    parser.add_argument(
+        '-p', 
+        '--port', 
+        dest='port', 
+        type=str,
```
Comment (cansubmarinesswim) [inline ml/demo_client.py]: Maybe int?

**Comment by cansubmarinesswim** [comment]
Is it actually the proper way of implementing ML component based on our proposed architecture? Maybe i'm missreading this, but it's one service at the moment - the `prediction model controller service`  and the `prediction model serivice` merged into one.

File: ml/src/app.py (hunk 1 above)
Comment (mstopa) [reply ml/src/app.py:43]: what about
```
    try:
        _ = _classify("Input text")[0][0]
        return True, "healthy"
    except:
        return False, "service is broken"
```

File: ml/demo_client.py (hunk 2 above)
Comment (mstopa) [reply ml/demo_client.py]: done

File: ml/.dockerignore (hunk 3 above)
Comment (mstopa) [reply ml/.dockerignore]: done

**Comment by mstopa** [comment]
https://github.com/cansubmarinesswim/posts-analyser/pull/4#issuecomment-1304852140

Well we might want to update the architecture proposal. There seems to be no point in separating a special controller for ML, because the model itself doesn't do much more than the controller does.

File: ml/src/app.py (hunk 1 above)
Comment (cansubmarinesswim) [reply ml/src/app.py:43]: Sounds fine!

File: ml/Dockerfile (hunk 4 above)
Comment (mstopa) [reply ml/Dockerfile]: done

File: ml/Dockerfile (hunk 5 above)
Comment (mstopa) [reply ml/Dockerfile:13]: done

File: ml/demo_client.py (hunk 6 above)
Comment (mstopa) [reply ml/demo_client.py]: done

**Pull request merged**